### PR TITLE
Add return_exceptions to litellm.batch_completion

### PR DESCRIPTION
**Pull request by ffreemt**
This PR adds a kwarg ``return_exceptions`` to litellm.batch_completion. When `return_exceptions` is set to False (default),  `batch_completion` behaves exactly as before. But when batch_completion is set to `True`, batch_completion will return exceptions and partial resuslt instead of throwing exceptions.

The idea of `return_exceptions` is borrowed from `asyncio.gather`. `batch_completion` makes use of `concurrent.futures.ProcessPoolExecutor to run multiple copies of `litellm.completion`.  However, `litellm.completion` may fail and throw exceptions. When this happens, the patched `batch_completion` will return partial results when `return_exceptions` is set to True. 

   ``return_exceptions`` can be useful in many scenarios.

### CODE REVIEW

File: litellm/main.py [hunk 1]
```diff
@@ -2361,7 +2362,17 @@ def chunks(lst, n):
                     completions.append(future)
 
         # Retrieve the results from the futures
-        results = [future.result() for future in completions]
+        # results = [future.result() for future in completions]
+        if return_exceptions:
```
Comment (ishaan-jaff) [inline litellm/main.py:2369]: this is an undefined variable - you're never reading it from `kwargs`

**Review by ishaan-jaff — CHANGES_REQUESTED** [review]
please add this to docs too https://docs.litellm.ai/docs/completion/batching

**Comment by ffreemt** [comment]
Somehow one line (`return_exceptions: bool = False,`) has gone missing in the shuffling. I've added the line back and also added a `test` (litellm/litellm/tests/test_batch_completion_return_exceptions.py)

I am not sure how to add docs to  https://docs.litellm.ai/docs/completion/batching. But the docs part is present in the `batch_completion` function definition.
```
return_exceptions (bool): Whether to return exceptions and partial results when exceptions occur. Defaults to False.
```

**Review by ishaan-jaff — APPROVED** [review]
lgtm !

**Comment by ishaan-jaff** [comment]
This failed our CI/CD on your test @ffreemt 
CI/Cd log: https://app.circleci.com/pipelines/github/BerriAI/litellm/9278/workflows/ad7ac3db-499d-41a8-9128-0c7cdf9aa041/jobs/19795/tests#failed-test-0

```
Failed: An error occurred: OpenAIException - Error code: 400 - {'error': {'message': 'Unrecognized request argument supplied: return_exceptions', 'type': 'invalid_request_error', 'param': None, 'code': None}}
```

**Comment by ishaan-jaff** [comment]
@ffreemt reverting this merge - happy to merge again once the errors are fixed 
After the fixes, Can you add a screenshot of batch completions working for you locally with and without setting `return_exceptions`

**Comment by ffreemt** [comment]
There is the pytest test at https://github.com/BerriAI/litellm/pull/3397/commits/a7ec1772b1457594d3af48cdcb0a382279b841c7#diff-44852387ceb00aade916d6b314dfd5d180499e54f35209ae9c07179febe08b4b showing `batch completions are working for me locally with and without setting return_exceptions`.

I'll have a look and see if I can make that CI/CD  happy.

**Comment by ishaan-jaff** [comment]
thanks, I appreciate it @ffreemt !

**Pull request merged**